hail_search/hail_search_query.py

```diff
@@ -0,0 +1,322 @@
+        enums = ht_globals.pop('enums')
+
+        annotation_fields = {
+            'populations': lambda r: hl.struct(**{
+                population: self.population_expression(r, population) for population in self.POPULATIONS.keys()
+            }),
+            'predictions': lambda r: hl.struct(**{
+                prediction: hl.array(enums[path[0]][path[1]])[r[path[0]][f'{path[1]}_id']]
+                if enums.get(path[0], {}).get(path[1]) else r[path[0]][path[1]]
+                for prediction, path in self.PREDICTION_FIELDS_CONFIG.items()
+            }),
+            'transcripts': lambda r: hl.or_else(
+                r.sorted_transcript_consequences, hl.empty_array(r.sorted_transcript_consequences.dtype.element_type)
+            ).map(
+                lambda t: self._enum_field(t, enums['sorted_transcript_consequences'], **self._format_transcript_args())
+            ).group_by(lambda t: t.geneId),
+        }
+        annotation_fields.update(self.BASE_ANNOTATION_FIELDS)
+
+        format_enum = lambda k, enum_config: lambda r: self._enum_field(r[k], enums[k], ht_globals=ht_globals, **enum_config)
+        annotation_fields.update({
+            enum_config.get('response_key', k): format_enum(k, enum_config)
+            for k, enum_config in self.ENUM_ANNOTATION_FIELDS.items()
+        })
+
+        if self._genome_version == GENOME_VERSION_GRCh38_DISPLAY:
+            annotation_fields.update(self.LIFTOVER_ANNOTATION_FIELDS)
+        return annotation_fields
+
+    def population_expression(self, r, population):
+        pop_config = self._format_population_config(self.POPULATIONS[population])
+        pop_field = self.POPULATION_FIELDS.get(population, population)
+        return hl.struct(**{
+            response_key: hl.or_else(r[pop_field][field], '' if response_key == 'id' else 0)
+            for response_key, field in pop_config.items() if field is not None
+        })
+
+    def _format_transcript_args(self):
+        return {
+            'format_value': lambda value: value.rename({k: _to_camel_case(k) for k in value.keys()}),
+        }
+
+    @staticmethod
+    def _enum_field(value, enum, ht_globals=None, annotate=None, format_value=None, drop_fields=None, **kwargs):
+        annotations = {}
+        drop = [] + (drop_fields or [])
+        value_keys = value.keys()
+        for field, field_enum in enum.items():
+            is_array = f'{field}_ids' in value_keys
+            value_field = f"{field}_id{'s' if is_array else ''}"
+            drop.append(value_field)
+
+            enum_array = hl.array(field_enum)
+            if is_array:
+                annotations[f'{field}s'] = value[value_field].map(lambda v: enum_array[v])
+            else:
+                annotations[field] = enum_array[value[value_field]]
+
+        value = value.annotate(**annotations)
+        if annotate:
+            annotations = annotate(value, enum, ht_globals)
+            value = value.annotate(**annotations)
```

Comment:
The term `annotation` is used for both variant search and hail. It is confusing. Should we use `variant_annotation` or another name for the search filter?